File: mavenSampleProject-kbtool-min/src/main/java/com/jch/jpa/JpaEntity.java
package com.jch.jpa;

import lombok.Data;
import lombok.experimental.Accessors;
import org.springframework.data.jpa.domain.AbstractAuditable;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.UUID;

@Entity
@Data
@Accessors(chain = true)
@Table(name = "t_jpa")
public class JpaEntity extends AbstractAuditable<String,UUID> {
	@GeneratedValue
	@Id
	UUID uuid;
	String name;
}

File: mavenSampleProject/src/main/java/org/apache/myfaces/JpaEntity.kt
package org.apache.myfaces

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Component
import javax.persistence.*

interface MyJpaReposotory : JpaRepository<JpaEntity, String>

@Entity
@Component
@Table(name = "tag_camera_tag_detail")
open class JpaEntity {
	@get:Id
	@get:Column(name = "index_code", nullable = false)
	var indexCode: String? = null

	@get:Basic
	@get:Column(name = "tag_list", nullable = true)
	var tagList: String? = null


	override fun toString(): String =
		"Entity of type: ${javaClass.name} ( " +
				"indexCode = $indexCode " +
				"tagList = $tagList " +
				")"

	// constant value returned to avoid entity inequality to itself before and after it's update/merge
	override fun hashCode(): Int = 42

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (javaClass != other?.javaClass) return false
		other as JpaEntity

		if (indexCode != other.indexCode) return false
		if (tagList != other.tagList) return false

		return true
	}

}

